### PR TITLE
Dashboard mail

### PR DESCRIPTION
**Pull request by Darshan2003**
## Summary
 Target issue is #2967  

## Notes
 Please add here if any other information is required for the reviewer. 
- Reused the SVGs generating for the live dashboard
- Created a basic structure for sending emails with dashboard graphs, sending it as HTML emails
- Working on creating a settings column in the organization table, which would give the frequency of the report to be sent

### CODE REVIEW

File: lib/glific/communications/mailer.ex [hunk 1]
```diff
@@ -76,6 +76,34 @@ defmodule Glific.Communications.Mailer do
...
+
+    in_cc = in_cc ++ [glific_support()]
+
+    new()
+    |> to(send_to)
+    |> from(from_email)
+    |> cc(in_cc)
+    |> subject(subject)
+    |> html_body(body)
+  end
```
Comment (AkhileshNegi) [inline lib/glific/communications/mailer.ex]: @Darshan2003 you can merge both the functions `common_html_send` and `common_send` and pass additional opts to `common_send` as send_html as true and handle it separately

File: lib/glific/reports.ex [hunk 2]
```diff
@@ -151,14 +151,36 @@ defmodule Glific.Reports do
...
+    start_day = case duration do
+      "MONTHLY" -> day |> Date.beginning_of_month()
+      "WEEKLY" -> day |> Date.beginning_of_week()
+      "DAILY" -> day
+    end
+
+    end_day = case duration do
+      "MONTHLY" -> day |> Date.end_of_month()
+      "WEEKLY" -> day |> Date.end_of_week()
+      "DAILY" -> day
```
Comment (AkhileshNegi) [inline lib/glific/reports.ex]: you can create a single function that matches the period and returns all three things period, start_day and end_day based on that in tuple like {period, start_day, end_day}

File: lib/glific_web/templates/dashboard/dashboard.html.heex [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    div {margin:0 !important;}
+	</style>
+  <noscript>
+    <xml>
+      <o:OfficeDocumentSettings>
+        <o:PixelsPerInch>96</o:PixelsPerInch>
+      </o:OfficeDocumentSettings>
+    </xml>
+  </noscript>
+  <![endif]-->
```
Comment (AkhileshNegi) [inline lib/glific_web/templates/dashboard/dashboard.html.heex]: remove commented code

File: lib/glific_web/templates/dashboard/dashboard.html.heex [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+            <tr>
+              <td style="background-color:#ffffff;">
+
+                <div class="two-col" style="text-align:center;font-size:20px;">
+                  <!--[if mso]> 
+        <table role="presentation" width="100%"> 
+        <tr> 
+        <td style="width:50%;" valign="middle"> 
+        <![endif]-->
```
Comment (AkhileshNegi) [inline lib/glific_web/templates/dashboard/dashboard.html.heex]: remove commented code

File: lib/glific_web/templates/dashboard/dashboard.html.heex [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+                      style="padding: 30px 10px;width: 350px; height: 250px; background-color: #f0f8f0; border-radius: 10px; box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); ">
+
+                      <%= @conversation_chart_svg %>
+                    </div>
+                  </div>
+                  <!--[if mso]> 
+        </td> 
+        </tr> 
+        </table> 
+        <![endif]-->
```
Comment (AkhileshNegi) [inline lib/glific_web/templates/dashboard/dashboard.html.heex]: remove commented code

File: lib/glific_web/templates/dashboard/dashboard.html.heex [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+            
+            <tr>
+              <td style="background-color:#ffffff;">
+
+                <div class="two-col" style="text-align:center;font-size:20px;">
+                  <!--[if mso]> 
+        <table role="presentation" width="100%"> 
+        <tr> 
+        <td style="width:50%;" valign="middle"> 
+        <![endif]-->
```
Comment (AkhileshNegi) [inline lib/glific_web/templates/dashboard/dashboard.html.heex]: remove commented code

File: lib/glific_web/templates/dashboard/dashboard.html.heex [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+                  <div class="column" style="padding-left: 10px;padding-top:10px; width:100%;max-width:390px;display:inline-block;vertical-align:middle;">
+                    <div id="optin-chart" class="card"
+                      style="padding: 30px 10px; width: 350px; height: 250px; background-color: #f0f8f0; border-radius: 10px; box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); ">
+                      <%= @optin_chart_svg %>
+                    </div>
+                  </div>
+                  <!--[if mso]> 
+        </td> 
+        <td style="width:50%;"  valign="middle"> 
+        <![endif]-->
```
Comment (AkhileshNegi) [inline lib/glific_web/templates/dashboard/dashboard.html.heex]: remove commented code

File: lib/glific_web/templates/dashboard/dashboard.html.heex [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+                    <div id="message-chart" class="card"
+                      style="padding: 30px 10px;width: 350px; height: 250px; background-color: #f0f8f0; border-radius: 10px; box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); ">
+                      <%= @message_chart_svg %>
+                    </div>
+                  </div>
+                  <!--[if mso]> 
+        </td> 
+        </tr> 
+        </table> 
+        <![endif]-->
```
Comment (AkhileshNegi) [inline lib/glific_web/templates/dashboard/dashboard.html.heex]: remove commented code

**Review by AkhileshNegi — CHANGES_REQUESTED** [review]

File: lib/glific_web/templates/dashboard/dashboard.html.heex (hunk 3 above)
Comment (Darshan2003) [reply lib/glific_web/templates/dashboard/dashboard.html.heex]: It is a conditional code which will only be rendered only on Outlook mail, It is needed for the mail to be responsive on outlook as well.

For reference: https://webdesign.tutsplus.com/creating-a-future-proof-responsive-email-without-media-queries--cms-23919t

File: lib/glific_web/templates/dashboard/dashboard.html.heex (hunk 4 above)
Comment (Darshan2003) [reply lib/glific_web/templates/dashboard/dashboard.html.heex]: It is a conditional code which will only be rendered only on Outlook mail, It is needed for the mail to be responsive on outlook as well.

For reference: https://webdesign.tutsplus.com/creating-a-future-proof-responsive-email-without-media-queries--cms-23919t

File: lib/glific_web/templates/dashboard/dashboard.html.heex (hunk 5 above)
Comment (Darshan2003) [reply lib/glific_web/templates/dashboard/dashboard.html.heex]: It is a conditional code which will only be rendered only on Outlook mail, It is needed for the mail to be responsive on outlook as well.

For reference: https://webdesign.tutsplus.com/creating-a-future-proof-responsive-email-without-media-queries--cms-23919t

File: lib/glific_web/templates/dashboard/dashboard.html.heex (hunk 6 above)
Comment (Darshan2003) [reply lib/glific_web/templates/dashboard/dashboard.html.heex]: It is a conditional code which will only be rendered only on Outlook mail, It is needed for the mail to be responsive on outlook as well.

For reference: https://webdesign.tutsplus.com/creating-a-future-proof-responsive-email-without-media-queries--cms-23919t

**Review by AkhileshNegi — APPROVED** [review]

**Pull request merged**